core/src/main/java/com/opensymphony/xwork2/config/providers/XmlConfigurationProvider.java

```diff
@@ -777,11 +777,21 @@ protected Class verifyResultType(String className, Location loc) {
                 }
                 params.putAll(resultParams);
 
-                ResultConfig resultConfig = new ResultConfig.Builder(resultName, resultClass)
-                        .addParams(params)
-                        .location(DomHelper.getLocationObject(element))
-                        .build();
-                results.put(resultConfig.getName(), resultConfig);
+                Set<String> resultNamesSet;
+                if (",".equals(resultName.trim())) {
+                    resultNamesSet = new HashSet<>(1);
+                    resultNamesSet.add(resultName);
```

Comment:
`if (",".equals(resultName.trim()))` assumes that the result name contains only `,` and then it will produce a set with `,` (value of `resultName`). `TextParseUtil.commaDelimitedStringToSet` is in the `else` section so it won't be used in this case.